__tests__/app.test.js

```diff
@@ -142,7 +142,140 @@ describe('4. GET /api/articles/:article_id', ()=>{
         .expect(404)
         .then((response)=>{
             const msg = response.body.msg;
-            expect(msg).toBe('Article Not Found');
+            expect(msg).toBe('Not Found');
+        })
+    });
+});
+
+describe('5. GET api/articles/:article_id/comments', ()=>{
+    test('responds with an array of comments for the given article_id when there are existing comments', ()=>{
+        return request(app)
+        .get('/api/articles/1/comments')
+        .expect(200)
+        .then(({body:{comments}})=>{
+            expect(comments).toHaveLength(11);
+            comments.forEach((comment)=>{
+                expect(comment).toMatchObject({
+                    comment_id: expect.any(Number),
+                    body: expect.any(String),
+                    votes: expect.any(Number),
+                    author: expect.any(String),
+                    created_at: expect.any(String),
+                })
+            })
+        })
+    });
+    test('responds with an empty array when given a valid article_id with no existing comment', ()=>{
+        return request(app)
+        .get('/api/articles/2/comments')
+        .expect(200)
+        .then(({body:{comments}})=>{
+            expect(comments).toHaveLength(0);
+        })
+    });
+    test('returns the array of comments sorted by date and is the most recent date(descending order) by default', ()=>{
+        return request(app)
+        .get('/api/articles/3/comments')
+        .expect(200)
+        .then(({body: { comments }})=>{
+            expect(comments).toBeSortedBy('created_at', {descending: true})
+        });
+    });
+
+    test('400: invalid article_id returns bad request', ()=>{
+        return request(app)
+        .get('/api/articles/monkey/comments')
+        .expect(400)
+        .then((response)=>{
+            const msg = response.body.msg;
+            expect(msg).toBe('Bad Request');
+        })
+    });
+    test('404: Valid but non-existent article_id returns not found', ()=>{
+        return request(app)
+        .get('/api/articles/100/comments')
+        .expect(404)
+        .then((response)=>{
+            const msg = response.body.msg;
+            expect(msg).toBe('Not Found');
+        })
+    });
+});
+
+describe('6. POST api/articles/:article_id/comments', ()=>{
+    test('201: responds with the newly posted comment', ()=>{
+        const newComment = {
+            username: "butter_bridge",
+            body: "my eyes hurt from looking at this article"
+        };
+        return request(app)
+        .post('/api/articles/6/comments')
+        .send(newComment)
+        .expect(201)
+        .then(({body})=>{
+            expect(body.comment[0]).toEqual({
```

Comment:
It's a bit less brittle if we use toMatchObject here or ObjectContaining. If we were to add columns to this table in the future this test currently would fail

Reply:
Alright, I'll work on this. Thank you